### DISCUSSION TITLE
one of `npm run build` or `npm start` for nextjs is not working properly on ec2 (ubuntu)

### DISCUSSION THREAD

**Discussion by mfaizan1** [vercel/next.js#31445] [Help]
### What version of Next.js are you using?

12.0.2

### What version of Node.js are you using?

v14.17.4

### What browser are you using?

irrelevant

### What operating system are you using?

ubuntu

### How are you deploying your application?

next build and next start on ec2

### Describe the Bug

I have a nextjs project, it builds and runs fine on locally but on ec2 server `yarn build`, `npm run build` and `npx next build` gives only this output

```
[ec2-user@ip-xxx-xx-xx-xx project-name]$ yarn build
yarn run v1.22.17
$ next build
info  - Loaded env from /home/ec2-user/project-name/.env
info  - Checking validity of types  
Done in 98.81s.
```
although there are no addtional logs like for local build it creates `.next` folder and when i try `npm start` or `yarn start` it throws this error

```
[ec2-user@ip-xxx-xx-xx-xx project-name]$ yarn start
yarn run v1.22.17
$ next start
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /home/ec2-user/project-name/.env
Error: Could not find a production build in the '/home/ec2-user/project-name/.next' directory. Try building your app with 'next build' before starting the production server
. https://nextjs.org/docs/messages/production-start-no-build-id
    at Server.readBuildId (/home/ec2-user/project-name/node_modules/next/dist/server/next-server.js:1825:23)
    at new Server (/home/ec2-user/project-name/node_modules/next/dist/server/next-server.js:100:29)
    at NextServer.createServer (/home/ec2-user/project-name/node_modules/next/dist/server/next.js:105:16)
    at async /home/ec2-user/project-name/node_modules/next/dist/server/next.js:117:31
error Command failed with exit code 1.
```
i am unable to figure out what the problem is as there are no problematic logs from build commands, only error i get seems like false error because `.next` folder is being created

### Expected Behavior

it should build and start correctly

### To Reproduce

I am not sure how to reproduce this as i created new next app and was able to build and run it on same ec2 instance, so problem is local to my project but logs are not helpful to debug my issue.

**Comment by timneutkens** [comment]
Converted to discussions as no code was provided. One reason I could think of is that you have customized `distDir` in next.config.js.

**Reply by mfaizan1** [reply]
@timneutkens  thanks for replying but i think i never made any change in `nextjs.config.js`

Here are contents of `next.config.js`
```
/** @type {import('next').NextConfig} */
module.exports = {
  reactStrictMode: true,
}
```

**Reply by designskiller** [reply]
@mfaizan1 did you get it resolve?

**Reply by mfaizan1** [reply]
> @mfaizan1 did you get it resolve?
Sorry i don't remember now.

**Reply by bbergeron-xlkey** [reply]
> you have customized distDir in next.config.js

That was part of my issue ( https://github.com/vercel/next.js/discussions/69740). That's a good start for anyone encountering a similar issue.
